Add a Cloudflare DNS Validation Plugin

georg-jung opened the pull request:
I created a plugin for dns-01 validation using Cloudflare. I tried to think of every place that needs to be changed (build scripts, docs, etc) and used many concepts of the existing azure plugin. I hope you think this is in scope for an additional plugin. I'd be happy to implement any changes or improvements to this if you see anything that should be done differently. Due to the simplicity of the CF API what this does seems simpler than what is required for i.e. Azure and the setup process should be quite straight forward for the users.

Thank you for providing and constantly improving win-acme!

georg-jung commented:
If you want to I can privately send you an API Token that provides DNS read and write access to the cloudflare-api-debug.com zone for testing purposes.

WouterTinus commented:
This looks very well out together Georg, thanks for the contribution! Would you mind to rebase the PR on the 2.1.3 branch? If I merge to master then the website would already get updated before the binaries are out. 

You can send the credentials to win.acme.simple@gmail.com, I'd be happy to give it a try.

georg-jung commented:
* I rebased onto 2.1.3 and my test afterwards worked fine.
  * Note that your commit f0e0bc48ee5556f9ac31181943fabcd577b040b5 is in master but not in 2.1.3
* I noticed I forgot to add my docs md to the `docs.csproj`, fixed this
* The information in my docs regarding how to configure the api token was inaccurate, fixed this.
